src/__internal__/full-screen-heading/full-screen-heading.component.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import tagComponent from '../../utils/helpers/tags';
+import StyledFullScreenHeading, { StyledHeaderContainer } from './full-screen-heading.style';
+import IconButton from '../../components/icon-button';
+import Icon from '../../components/icon';
+
+const FullScreenHeading = React.forwardRef((props, ref) => {
+  const {
+    children,
+    showCloseIcon,
+    onCancel,
+    ...otherProps
+  } = props;
+
+  const closeIcon = () => {
+    if (!showCloseIcon || !onCancel) return null;
+    return (
+      <IconButton
+        data-element='close'
+        onAction={ onCancel }
```

Comment:
This prop is required in `IconButton` but you might end up passing `undefined` here

Reply:
I'll put something defensive in here to stop that from happening

Reply:
Actually the line above `if (!showCloseIcon || !onCancel) return null;` already does that, it's not possible to pass `undefined` through here

Reply:
Looks like it's checking the prop type erroneously then, as this component is going to be refactored anyway we might as well just look to fix it then as the issue is fairly trivial